packages/@okta/vuepress-site/docs/guides/mobile-idx-sdk-overview/main/ios/signingout.md

```diff
@@ -1,94 +1,18 @@
-This singleton stores the token in user defaults. For more security, store the token in the Keychain.
+The code snippet below uses the `Credential` object of the `AuthFoundation` library to revoke the tokens for the current user. You can use other features of `Credential` to revoke the tokens of a specific user.
 
 ```swift
 
-import Foundation
 import OktaIdx
+import AuthFoundation
 
-enum TokenManagerError: Error {
-    case unableToReadToken(String)
-}
-
-class TokenManager {
-    static let shared = TokenManager()
-
-    // Change com.example.okta-samplecode.Okta-Login to your bundle identifier.
-    private struct UserDefaultsKeys {
-        static let storedTokenKey = "com.example.okta-samplecode.Okta-Login.storedToken"
-    }
-
-    // Save the token to user defaults.
-    func saveToken(_ token: Token) {
-        let defaults = UserDefaults.standard
-
-        // Convert the token to a format that you can save in user defaults.
-        guard let tokenData = try? JSONEncoder().encode(token) else {
-            // Handle the error.
-            return
-        }
-
-        defaults.set(tokenData, forKey: UserDefaultsKeys.storedTokenKey)
-        defaults.synchronize()
-    }
-
-    // Read the token from user defaults.
-    func readToken() -> Token? {
-        // Check for an existing token and convert from JSON to an SDK Token object.
-        guard let tokenData = UserDefaults.standard.object(forKey: UserDefaultsKeys.storedTokenKey) as? Data,
-              let token = try? JSONDecoder().decode(Token.self, from: tokenData) else {
-            // Handle the error.
-            return nil
-        }
-
-        return token
-    }
-
-    func revokeToken(completion: @escaping(_ successful: Bool, _ error: Error?) -> Void) {
-        if let token = readToken() {
-            token.revoke() { (success, error) in
-               let defaults = UserDefaults.standard
-               if success {
-                    // Delete the token from user defaults.
-                    defaults.removeObject(forKey: UserDefaultsKeys.storedTokenKey)
-                    defaults.synchronize()
-
-                    completion(true, nil)
-                } else {
-                    completion(false, error)
-                }
-            }
-        } else {
-            completion(false, TokenManagerError.unableToReadToken("Unable to read the Token."))
-            return
-        }
-    }
-}
-```
-
-This code shows the delegate function that handles receiving the access token:
+class SignInController: InteractionCodeFlowDelegate, ObservableObject {
+//   ...
 
-
-```swift
-public func idx(client: IDXClient, didReceive token: Token) {
-    TokenManager.shared.save(token)
+   // Sign out the current user.
+   public func signOut() throws {
+      try Credential.default?.remove()
```

Comment:
I'd suggest you use `try await Credential.default?.revoke()` instead.

Reply:
Thanks. Update is in the next commit.